src/js/components/SchemaForm.js

```diff
@@ -107,7 +107,17 @@ class SchemaForm extends mixin(StoreMixin) {
       propID,
       newDefinition
     );
-    newDefinition.push(
+
+    let deleteButtonTop = Object.values(definition.itemShapes || {})
```

Comment:
what if we put `deleteButtonTop` on the definition itself, rather than on one of it's itemShapes